### ISSUE TITLE
Can I extract mathematical formulation?

### ISSUE THREAD

**Issue by xkhainguyen** [johannahaffner/sif2jax#49]
Hi,

I wanted to use CUTEst as a standardized benchmark for my machine learning-based approaches using Python. However, calling Fortran routines sequentially on CPUs is inefficient.

To address this, I believe I need to know the underlying mathematical formulation (symbolic expressions of the objective and constraints) for each problem so that I can implement them natively and perform batched evaluations and gradient backpropagation on GPUs.

Do you know of any ways to do this? Also feel free to suggest any alternatives.

Many thanks!

**Comment by johannahaffner** [comment]
Sif2jax does not call Fortran at all, it is a reimplementation of CUTEst problems that is JAX-native, mostly written in pure Python but does use the occasional `jax.vmap` in some functions. 

Given that, you might be able to use it directly?
